src/data/customers/customerTestCases.data.ts

```diff
@@ -0,0 +1,212 @@
+    },
+    {
+        name: "20-char city",
+        data: generateCustomerData({ name: "A".repeat(5) + ' ' + "A".repeat(14) }),
+    },
+    {
+        name: "City with post/prefix spaces",
+        data: generateCustomerData({ city: " Stockholm " }),
+    },
+    {
+        name: "1-char street",
+        data: generateCustomerData({ street: "A" }),
+    },
+    {
+        name: "40-char street",
+        data: generateCustomerData({ name: "A".repeat(15) + ' ' + "A".repeat(24) }),
+    },
+    {
+        name: "Street with post/prefix spaces",
+        data: generateCustomerData({ street: " Mannerheim katu " }),
+    },
+    {
+        name: "House = 1",
+        data: generateCustomerData({ house: 1 }),
+    },
+    {
+        name: "House = 999",
+        data: generateCustomerData({ house: 999 }),
+    },
+    {
+        name: "Flat = 1",
+        data: generateCustomerData({ flat: 1 }),
+    },
+    {
+        name: "Flat = 9999",
+        data: generateCustomerData({ flat: 9999 }),
+    },
+    {
+        name: "Min valid phone",
+        data: generateCustomerData({ phone: "+1234567890" }),
+    },
+    {
+        name: "Max length phone",
+        data: generateCustomerData({ phone: "+12345678909876543210" }),
+    },
+    {
+        name: "Phone with post/prefix spaces",
+        data: generateCustomerData({ phone: " +1234567890  " }),
+    },
+    {
+        name: "250-char notes",
+        data: generateCustomerData({ notes: "N".repeat(250) }),
+    },
+    {
+        name: "Notes with post/prefix spaces",
+        data: generateCustomerData({ notes: " Some notes " }),
+    },
+    {
+        name: "Extra unknown field",
+        data: {
+            ...generateCustomerData(),
+            extraField: "should be ignored",
+        },
+    },
+];
+
+export const negativeTestCases = [
+    {
+        name: "Missing auth token",
+        data: generateCustomerData(),
+        expectedError: "Not authorized"
+    },
+    {
+        name: "Invalid auth token",
+        data: generateCustomerData(),
+        expectedError: "Invalid access token"
+    },
+    {
+        name: "Empty email",
+        data: generateCustomerData({ email: "" }),
+        expectedError: "Incorrect request body"
+    },
+    {
+        name: "Email without @",
+        data: generateCustomerData({ email: "invalidemail.com" }),
+        expectedError: "Incorrect request body"
+    },
+    {
+        name: "Missing email field",
+        data: generateCustomerData(),
+        override: (newCustomerData: ICustomer) => {
+            const { email, ...rest } = newCustomerData;
+            return rest;
+        },
+        expectedError: "Incorrect request body"
+    },
+    {
+        name: "Empty name",
+        data: generateCustomerData({ name: "" }),
+        expectedError: "Incorrect request body"
+    },
+    {
+        name: "Name too long (41 chars)",
+        data: generateCustomerData({ name: "A".repeat(30) + ' ' + "A".repeat(10) }),
```

Comment:
У нас есть небольшое правило
`Use template literals instead of concatenating strings.`
```suggestion
        data: generateCustomerData({ name: "A".repeat(30) + ' ' + "A".repeat(10) }),
```